e2e/testsuite/testsuite.go

```diff
@@ -0,0 +1,242 @@
+package testsuite
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	dockerclient "github.com/docker/docker/client"
+	"github.com/strangelove-ventures/ibctest"
+	"github.com/strangelove-ventures/ibctest/chain/cosmos"
+	"github.com/strangelove-ventures/ibctest/ibc"
+	"github.com/strangelove-ventures/ibctest/testreporter"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+
+	"github.com/cosmos/ibc-go/v4/e2e/testconfig"
+)
+
+const (
+	// ChainARelayerName is the name given to the relayer wallet on ChainA
+	ChainARelayerName = "rlyA"
+	// ChainBRelayerName is the name given to the relayer wallet on ChainB
+	ChainBRelayerName = "rlyB"
+)
+
+// E2ETestSuite has methods and functionality which can be shared among all test suites.
+type E2ETestSuite struct {
+	suite.Suite
+	paths          map[string]path
+	logger         *zap.Logger
+	DockerClient   *dockerclient.Client
+	network        string
+	startRelayerFn func(relayer ibc.Relayer)
+}
+
+// path is a pairing of two chains which will be used in a test.
+type path struct {
+	chainA, chainB *cosmos.CosmosChain
+}
+
+// newPath returns a path built from the given chains.
+func newPath(chainA, chainB *cosmos.CosmosChain) path {
+	return path{
+		chainA: chainA,
+		chainB: chainB,
+	}
+}
+
+// SetupChainsRelayerAndChannel create two chains, a relayer, establishes a connection and creates a channel
+// using the given channel options. The relayer returned by this function has not yet started. It should be started
+// with E2ETestSuite.StartRelayer if needed.
+// This should be called at the start of every test, unless fine grained control is required.
+func (s *E2ETestSuite) SetupChainsRelayerAndChannel(ctx context.Context, channelOpts ...func(*ibc.CreateChannelOptions)) ibc.Relayer {
+	chainA, chainB := s.GetChains()
+	home, err := ioutil.TempDir("", "")
```

Comment:
Should we be using empty strings here? Is this the chain home directory?

Reply:
with empty strings we will just have a random string generated for the home directory. After this we don't need to interact with the home dir directly.